### PR TITLE
Attack Response Modification

### PR DESCRIPTION
**Pull request by saeidalz13**
# Overview

As per our discussion, attack response was modified.

- `DefenderSunkenShipsCoords` field was added to attack response.
- `HandleAttack` handler was modified.
- Minor bug fixes.
- Commented code moved away.

### CODE REVIEW

File: models/battleship/ship.go [hunk 1]
```diff
@@ -40,3 +42,7 @@ func (sh *Ship) GotHit() {
 func (sh *Ship) IsSunk() bool {
 	return sh.hits == sh.length
 }
+
+func (sh *Ship) GetHitCoords() []Coords {
+	return sh.coordsHits
```
Comment (mori-ahk) [inline models/battleship/ship.go:47]: nit: `hitCoordinates` is better naming

Comment (saeidalz13) [reply models/battleship/ship.go:47]: Ahhh you're right, thanks for being nit picky, I want it!

**Review by mori-ahk — COMMENTED** [review]
DO NOT MERGE THIS as it will break the app for TestFlight users. Please deploy on staging first. Once the client is sync with this new approach. we can ship it to prod

File: models/battleship/grid.go [hunk 2]
```diff
@@ -2,6 +2,15 @@ package battleship
 
 type GridInt [][]int
 
+type Coords struct {
```
[the remark this answers is not in the document]
Comment (mori-ahk) [reply models/battleship/grid.go:5]: @saeidalz13 you missed this one I think.

[the remark this answers is not in the document]
Comment (saeidalz13) [reply models/battleship/grid.go]: fixed

**Review by mori-ahk — APPROVED** [review]

**Pull request merged**